### ISSUE TITLE
Using sphinx.ext.imgmath raises error

### ISSUE THREAD

**Issue by pique-b** [sphinx-doc/sphinx#11897]
### Describe the bug

I have been trying to use Sphinx's ``math`` directive. As per the help page at https://www.sphinx-doc.org/en/master/usage/extensions/math.html, I have configured my project as under:

conf.py file
------------

Added extension:

```
extensions = ['sphinx.ext.autodoc',
              'sphinx.ext.imgmath', # For Math
              '...',
]

imgmath_image_format = 'svg'

imgmath_dvisvgm = 'dvisvgm' # Tried to add the full path also (/opt/local/bin/)
```

Others
------

As suggested on the above referred to page, I have installed ``dvisvgm`` using MacPorts.

### How to Reproduce

I have the following code in my ``.rst`` file:

```
.. math::
   W^{3\beta}_{\delta_1 \rho_1 \sigma_2} \approx U^{3\beta}_{\delta_1 \rho_1}

```

Using command ``make html``, raising the following error:

```
WARNING: inline latex 'W^{3\\beta}_{\\delta_1 \\rho_1 \\sigma_2} \\approx U^{3\\beta}_{\\delta_1 \\rho_1}\n\n': latex exited with error
[stderr]
Latexmk: This is Latexmk, John Collins, 26 Dec. 2019, version: 4.67.
Rule 'latex': The following rules & subrules became out-of-date:
      'latex'
------------
Run number 1 of rule 'latex'
------------
------------
Running 'latex  --interaction=nonstopmode -recorder  "math.tex"'
------------
Latexmk: Missing input file: 'anyfontsize.sty' from line
  '! LaTeX Error: File `anyfontsize.sty' not found.'
Latexmk: Log file says no output from latex
Latexmk: For rule 'latex', no output was made
Collected error summary (may duplicate other messages):
  latex: Command for 'latex' gave return code 1
      Refer to 'math.log' for details
Latexmk: Use the -f option to force complete processing,
 unless error was exceeding maximum runs, or warnings treated as errors.

[stdout]
Latexmk: applying rule 'latex'...
This is pdfTeX, Version 3.14159265-2.6-1.40.20 (TeX Live 2019) (preloaded format=latex)
 restricted \write18 enabled.
entering extended mode
(./math.tex
LaTeX2e <2019-10-01>
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/base/article.cls
Document Class: article 2019/08/27 v1.4j Standard LaTeX document class
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/base/size12.clo))
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/base/inputenc.sty)
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsmath/amsmath.sty
For additional information on amsmath, use the `?' option.
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsmath/amstext.sty
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsmath/amsgen.sty))
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsmath/amsbsy.sty)
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsmath/amsopn.sty))
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amscls/amsthm.sty)
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsfonts/amssymb.sty
(/usr/local/texlive/2019basic/texmf-dist/tex/latex/amsfonts/amsfonts.sty))

! LaTeX Error: File `anyfontsize.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
! Emergency stop.
<read *> 
         
l.8 \usepackage
               {bm}^^M
No pages of output.
Transcript written on math.log.
=== TeX engine is 'pdfTeX'
Latexmk: Errors, so I did not complete making targets

```

I can see that system detected a missing file **anyfontsize.sty**. In this situation I don't know what to do. Tried to find a solution but no straight forward solution is forthcoming.

### Environment Information

```text
System info:

* Python 3.6
* Sphinx 3.1.2
* Latexmk (Latex 2019 Basic) version 
* dvisvgm
```

### Additional context

I have tried the same process in **venv** with ``Python3.9`` and ``Sphinx 7.2.6``, but facing the same issue.

Trying to use ``sphinx-build`` like suggested here: https://www.sphinx-doc.org/en/master/usage/extensions/math.html#confval-imgmath_latex has not helped either:

``$ sphinx-build -M html -D imgmath_latex=latex source  build``

Raises error:

> WARNING: LaTeX command 'latex/html' cannot be run (needed for math display), check the imgmath_latex setting

If I build a **pdf** file from the doc, the formula is correctly displayed.

I am sorry if I have missed out on some additional information here. _Do kindly let me know._

**Under the circumstance could you please suggest a way around. What am I doing wrong?**

**Comment by jfbu** [comment]
The template for imgmath uses a few specific LaTeX dependencies: `anyfontsize` and `bm`.  You seem (from the quoted paths) to be using a LaTeX installation done via TeXLive and which selected only their "basic" scheme (see [texlive doc](https://tug.org/texlive/doc/texlive-en/texlive-en.html#x1-240003.2.2) for context). 

A priori executing in command line `tlmgr install anyfontsize` and (I did not check if `bm` is or not  in "basic" TeXLive "scheme") `tlmgr install bm` should solve the issue but you will most probably need to execute first something such as `tlmgr option repository="https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2019/tlnet-final/"` (**untested**) because the current TeXLive mirrors will complain if your install dates back from earlier than 2023.  See [this texlive doc](https://tug.org/historic/).

**Comment by AA-Turner** [comment]
@jfbu is there a cross-platform (read: Windows!) way of checking if a tex library is installed? On a user experience point, we could provide a better error message here, directing the reader to check their TeX installation.

A

**Comment by jfbu** [comment]
@AA-Turner cross-platform yes, as LaTeX has anyhow `\IfFileExists` which can be used from inside latex code.  The issue is cross-latexinstallationmethod. I am only familiar with TeXLive, which has the `kpsewhich` utility which returns path to a filename passed as argument, or `tlmgr info` which actually queries the package manager of TeXLive and will be able to provide the name to use in `tlmgr install` command, and whether it is installed or not.  No idea (actually I have a better idea now[^1]) about the analog with MikTeX, which was originally Windows only but can be used on other platforms, anyhow as far as I know MikTeX will fetch "on-the-fly" dependencies currently missing on the user system, during latex runs, so on Windows+MikTeX user would not have encountered the problem.

The error message here from Sphinx 

> > WARNING: LaTeX command 'latex/html' cannot be run (needed for math display), check the imgmath_latex setting

was partially misleading, but for once the voluminous console output from latex, inherited by Latexmk appears to be quite simple to understand.  We could possibly incorporate inside our warning the last ten lines of the `.log` file...

Before realizing user was with a TeXLive based install, I had checked the [Ubuntu](https://packages.ubuntu.com/) and `anyfontsize.sty` is in [texlive-latex-extra](https://packages.ubuntu.com/kinetic/texlive-latex-extra) which we [list officially](https://www.sphinx-doc.org/en/master/usage/builders/index.html#sphinx.builders.latex.LaTeXBuilder) as dependency.

Here the problem arose from [a latex imgmath template](https://github.com/sphinx-doc/sphinx/blob/7041f11fb00d85ecc7ba8951beca6726868ad50e/sphinx/templates/imgmath/template.tex_t) but I am actually somewhat surprised that the "basic" TeXLive scheme from user install proved sufficient to run Sphinx latexpdf builder: if true, this is a testimony to the fact that we use only a limited and cautiously curated collection of LaTeX packages.  For example we even check if `xcolor.sty` is available but no reasonable real-life installation of LaTeX can be without `xcolor` (e.g. many LaTeX users use TikZ/pgf which requires it).

To sum up, maybe we could add a check, but it adds overhead and in the case at hand the error message from latex itself was good, so I am not much in favour, I prefer we try to fetch the tail of the latex log file and incorporate it to our Sphinx issued warning. 

We could try to make an effort [in our docs](https://www.sphinx-doc.org/en/master/usage/builders/index.html#sphinx.builders.latex.LaTeXBuilder) to explain situation on Windows, but I don't myself have easy access to it so can't provide the information.

[^1]: actually [our test suite](https://github.com/sphinx-doc/sphinx/blob/7041f11fb00d85ecc7ba8951beca6726868ad50e/tests/test_builders/test_build_latex.py#L32-L37) uses the `kpsewhich` utility, which seems to indicate it must be also present in MikTeX based installations.  And yes I see it is from [this](https://tex.stackexchange.com/questions/305882/kpsewhich-returns-blank-line) for example.

**Comment by pique-b** [comment]
Thanks so much for taking time to delve on the issue.

Going through some of the docs available as well as help available on the web, surmised that it is better to upgrade Tex Live. However ``tlmgr`` was not of much help which kept insisting I update from my **legacy** install.

My *2019* install experience was quite messy (due to my older macOS version). So thought I would try installing it using MacPorts. The path I chose was:

- Selected the port ``texlive-basic`` and upon trying to compile found that the problem of the missing ``anyfontsize.sty`` persisted.

- Searching for the missing ``anyfontsize.sty`` finally led me to https://trac.macports.org/wiki/TeXLivePackages which listed ``anyfontsize.sty`` as one of the files of port ``texlive-latex-extra``. After installing ``texlive-latex-extra`` and on subsequent compiling, the error was no longer pointing to this particular file (yes this time too even after upgrading 2023).

- Instead the error now was related to file ``newtxsf.sty``.  So the next step was obvious. Installed port ``texlive-fonts-extra`` which was listed as having the missing file ``newtxsf.sty``. And on the next compile, the formulae showed up exactly as expected.

@jfbu  Thanks for your comments. 

> the current TeXLive mirrors will complain

Sure enough tried ``tlmgr`` and I kept getting message in the terminal that ``tlmgr`` can not be run and asked me to upgrade.

@AA-Turner 

> anyfontsize.sty is in [texlive-latex-extra](https://packages.ubuntu.com/kinetic/texlive-latex-extra) which we [list officially](https://www.sphinx-doc.org/en/master/usage/builders/index.html#sphinx.builders.latex.LaTeXBuilder) as dependency.

**Comment by jfbu** [comment]
> However `tlmgr` was not of much help which kept insisting I update from my **legacy** install.

Did you try `tlmgr option repository="https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2019/tlnet-final/"`? As per the word *legacy* in the TeXLive world on macOS it refers usually to the fact that the MacTeX (which is that section of TeXLive especially packaged fro macOS) only supports the N, N-1, and N-2 system iterations, but fortunately, one of the TeXLive volunteer compiles a version of binaries supporting much older mac os systems, and this is called `darwin-legacy`.   I was not aware TeXLive would use *legacy* in another acceptation, i.e. to refer that TeXLive insists you reinstall it architecture every year[^1] and if you don't the `tlmgr` will not update any LaTeX package... except if you point it to one of the few repositories holding "historic" past versions, as I indicated at start of this paragraph.

But I feel you have solved your problems via a MacPorts install anyhow.

[^1]: it is possible to do a rolling install, but this reserved to adventurous hackers and barely documented by TeXLive management.

**Comment by pique-b** [comment]
@jfbu 

> Did you try tlmgr option repository="https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2019/tlnet-final/"?

No, as I was not aware of it.

When I said *legacy* I had meant Tex Live install from tug.org. I had installed it circa 2020 but had to get the installer from some (I don't remember which) of their archive. Like I have mentioned above, it wasn't a nice experience installing it what with many dependencies having to be installed manually (I don't remember how did I actually do it!).

On the official Latex site I read ([about upgrading from 2022 rel. to 2023](https://www.tug.org/texlive/upgrade.html) and they have taken pains to explain why doing an upgrade is not recommended and that it should be left to the experienced Unix admins. *So tried to find an alternative route*.

I came across the *MacPorts* option only this time. *Fortuitously!!*

Thanks for giving the backgrounds. It is now pushing me to read up on the subject truly.

@AA-Turner :

> I am actually somewhat surprised that the "basic" TeXLive scheme from user install proved sufficient to run Sphinx latexpdf builder

My intention of taking resort to Sphinx for document generation was for one project I was trying to complete around the time I installed it. Like I have said in these paragraphs, I somehow managed to complete the docs part, put it in the project for onlineline access and also created a pdf as an offline help source.

Just to clarify, here are some of the images of the ``math`` directive usage (which were failing but compiled perfectly in ``latexpdf``):

![math_formula_1](https://github.com/sphinx-doc/sphinx/assets/69836231/3dcd78c4-abff-4c28-9e71-55261f6d27be)

![math_formula_pytho](https://github.com/sphinx-doc/sphinx/assets/69836231/6b14e663-8cea-400d-b981-509e6a8d04c9)

... and kudos to the brains behind **Sphinx** project which has as yet, many unknown and unexplored gems hidden under its myriad layers of codes.
